### PR TITLE
Make SonarAnalyzer optional

### PR DESCRIPTION
**Pull request by earlbread**
This is another version of #977.

This turns off Sonar Analyzer by default and checks it using in the `check-sonar` action.
In your local environment, you can skip Sonar by using `SkipSonar` option like `dotnet build -p:SkipSonar=true`.

### CODE REVIEW

**Review by longfin — DISMISSED** [review]

**Review by dahlia — COMMENTED** [review]
How about turning on the check by default, and turning it off on the CI?

**Review by dahlia — COMMENTED** [review]
It would be great if we mention `SkipSonar` in the *CONTRIBUTING.md* file.

**Comment by limebell** [comment]
How about make a new CI workflow that builds with Sonar Analyzer without running tests? Then we can make all test workflows to skip Sonar Analyzer.

**Review by riemannulus — DISMISSED** [review]

**Comment by earlbread** [comment]
I changed to turn off Sonar on CI and check it on `check-build` action.

File: .github/workflows/check-build.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - uses: actions/checkout@main
+      if: github.event_name != 'pull_request'
+    - uses: actions/checkout@main
+      if: github.event_name == 'pull_request'
+      with:
+        ref: ${{ github.pull_request.head.sha }}
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.100
+    - run: 'dotnet build'
```
Comment (limebell) [inline .github/workflows/check-build.yaml:20]: How about clarify using of Sonar Analyzer?
```suggestion
    - run: 'dotnet build p:SkipSonar=false'
```

File: .github/workflows/check-build.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+on:
```
Comment (dahlia) [inline .github/workflows/check-build.yaml:1]:
```suggestion
# This workflow checks if the build (compilation) succeeds without any errors.
# Although the build is done in Azure Pipelines as well, to speed up the build time
# some checks are turned off in Azure Pipelines.  To conduct the complete checks
# there should be this separated workflow.
# See also the below issues:
# - https://github.com/planetarium/libplanet/pull/979
# - https://github.com/planetarium/libplanet/pull/977
# - https://github.com/planetarium/libplanet/issues/976
on:
```

**Comment by dahlia** [comment]
After this pull request being merged, we should make the introduced **check-build** required in the branch protection settings.

**Review by dahlia — APPROVED** [review]
:+1:

**Comment by dahlia** [comment]
@planetarium/libplanet Please review this again!

**Review by longfin — APPROVED** [review]

**Pull request merged**